autopush/router/fcm_v1.py

```diff
@@ -15,6 +18,13 @@
 )
 from autopush.types import JSONDict  # noqa
 
+# the universal default for this is 4096, which is far too
+# large for FCM. The final payload size of the encoded data, plus
+# encryption headers must fit in the 4096 byte FCM payload.
+# Since the body is re-encoded base64, we reduce the message
+# size accordingly
+MAX_FCM_DATA = 3015
```

Comment:
It is more straightforward to implement the max data check in the FCM client instead of the router (See https://github.com/mozilla-services/autopush-rs/pull/212). The client has the full payload available. Then there's no need to use any limit other than 4096.

Reply:
I debated that, but part of the thing was I wanted to be helpful to the original sender. To do that, I'd need to have the original message data. I could deconstruct it from inside the sender client, or I could do it while I was composing the message to send through. I figured it was clearer to do it while composing. 

I could see an argument about putting the final message size check inside of the sender, and check the length of `payload.data_message` but it's just as straight forward to do it in the caller, and doesn't include a stack shift.

Reply:
It's not very clear, but from testing it seems that FCM only considers the `data` field when calculating the size of the payload, so this code is accurate enough.